include/crocoddyl/multibody/costs/contact-wrench-cone.hpp

```diff
@@ -0,0 +1,158 @@
+
+#ifndef CROCODDYL_MULTIBODY_COSTS_CONTACT_WRENCH_CONE_HPP_
+#define CROCODDYL_MULTIBODY_COSTS_CONTACT_WRENCH_CONE_HPP_
+
+#include "crocoddyl/multibody/fwd.hpp"
+#include "crocoddyl/multibody/cost-base.hpp"
+#include "crocoddyl/multibody/contact-base.hpp"
+#include "crocoddyl/multibody/contacts/contact-3d.hpp"
+#include "crocoddyl/multibody/contacts/contact-6d.hpp"
+#include "crocoddyl/multibody/data/contacts.hpp"
+#include "crocoddyl/multibody/frames.hpp"
+#include "crocoddyl/multibody/wrench-cone.hpp"
+#include "crocoddyl/core/utils/exception.hpp"
+#include "crocoddyl/core/utils/deprecate.hpp"
+
+namespace crocoddyl {
+
+template <typename _Scalar>
+class CostModelContactWrenchConeTpl : public CostModelAbstractTpl<_Scalar> {
+ public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  typedef _Scalar Scalar;
+  typedef MathBaseTpl<Scalar> MathBase;
+  typedef CostModelAbstractTpl<Scalar> Base;
+  typedef CostDataContactWrenchConeTpl<Scalar> Data;
+  typedef StateMultibodyTpl<Scalar> StateMultibody;
+  typedef CostDataAbstractTpl<Scalar> CostDataAbstract;
+  typedef ActivationModelAbstractTpl<Scalar> ActivationModelAbstract;
+  typedef ActivationModelQuadTpl<Scalar> ActivationModelQuad;
+  typedef DataCollectorAbstractTpl<Scalar> DataCollectorAbstract;
+  typedef FrameForceTpl<Scalar> FrameForce;
+  typedef WrenchConeTpl<Scalar> WrenchCone;
+  typedef FrameWrenchConeTpl<Scalar> FrameWrenchCone;
+  typedef typename MathBase::Vector6s Vector6s;
+  typedef typename MathBase::VectorXs VectorXs;
+  typedef typename MathBase::MatrixXs MatrixXs;
+  typedef typename MathBase::MatrixX6s MatrixX6s;
+
+  CostModelContactWrenchConeTpl(boost::shared_ptr<StateMultibody> state,
+                                  boost::shared_ptr<ActivationModelAbstract> activation, const FrameWrenchCone& fref,
+                                  const std::size_t& nu);
+  CostModelContactWrenchConeTpl(boost::shared_ptr<StateMultibody> state,
+                                  boost::shared_ptr<ActivationModelAbstract> activation,
+                                  const FrameWrenchCone& fref);
+  CostModelContactWrenchConeTpl(boost::shared_ptr<StateMultibody> state, const FrameWrenchCone& fref,
+                                  const std::size_t& nu);
+  CostModelContactWrenchConeTpl(boost::shared_ptr<StateMultibody> state, const FrameWrenchCone& fref);
+  virtual ~CostModelContactWrenchConeTpl();
+
+  virtual void calc(const boost::shared_ptr<CostDataAbstract>& data, const Eigen::Ref<const VectorXs>& x,
+                    const Eigen::Ref<const VectorXs>& u);
+  virtual void calcDiff(const boost::shared_ptr<CostDataAbstract>& data, const Eigen::Ref<const VectorXs>& x,
+                        const Eigen::Ref<const VectorXs>& u);
+  virtual boost::shared_ptr<CostDataAbstract> createData(DataCollectorAbstract* const data);
+
+  DEPRECATED("Use set_reference<FrameWrenchConeTpl<Scalar> >()", void set_fref(const FrameWrenchCone& fref));
```

Comment:
Remove this DEPRECATED functions. This was done to avoid breaking the code of our users, however, this function is new, so let's remove it.

Note that you need to remove also the implementation of `set_fref` and `get_fref`.